external/avro/src/main/scala/org/apache/spark/sql/avro/AvroLogicalTypeCatalystMapper.scala

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package org.apache.spark.sql.avro
+
+import org.apache.avro.{LogicalType, Schema}
+
+import org.apache.spark.sql.avro.SchemaConverters.SchemaType
+
+/**
+ * Mapping interface between Catalyst struct type and Avro schemas
+ */
+trait AvroLogicalTypeCatalystMapper extends Serializable {
```

Comment:
This should probably be marked as `@Experimental`, but someone from the Spark team should confirm.